common/src/main/java/net/draycia/carbon/common/messages/CarbonMessageSource.java

```diff
@@ -151,7 +148,16 @@ private void reloadTranslations() throws IOException {
     private void readLocale(final Path localeDirectory, final Path localeFile, final Locale locale) {
         this.logger.info("Found locale {} ({}) in: {}", locale.getDisplayName(), locale, localeFile);
 
-        final Properties properties = new Properties();
+        final Properties properties = new Properties() {
+            @Override
+            public synchronized Set<Map.Entry<Object, Object>> entrySet() {
```

Comment:
I would prefer a static method for `Properties` -> `Set<Map.Entry<Object, Object>>`, and replace relevant usages of `Properties#entrySet` with it, instead of extending `Properties`.